parsec/api/protocol/handshake.py

```diff
@@ -256,13 +256,17 @@ def process_answer_req(self, req: bytes) -> None:
         # Retrieve the version used by client first
         data = handshake_answer_version_only_serializer.loads(req)
         client_api_version = data["client_api_version"]
-        if client_api_version.version == 2:
-            serializer = handshake_answer_serializer
-        elif client_api_version.version == 1:
+
+        # API version matching
+        self.backend_api_version, self.client_api_version = _settle_compatible_versions(
+            self.SUPPORTED_API_VERSIONS, [client_api_version]
+        )
+
+        # Use the correct serializer
+        if client_api_version.version == 1:
             serializer = apiv1_handshake_answer_serializer
         else:
-            # Imcompatible version !
-            raise HandshakeAPIVersionError(self.SUPPORTED_API_VERSIONS, [client_api_version])
+            serializer = handshake_answer_serializer
```

Comment:
what if `client_api_version.version > 3` ? I guess this is catched by `_settle_compatible_versions` (hence why it has been moved up), but it would be better to have a comment about that

Reply:
Exactly, I'm adding the comment